### PR TITLE
Allow artefact suffix names to be passed in as an argument

### PR DESCRIPTION
**Pull request by duncancrawford**
This change will optionally allow for a suffix to be passed in as an argument parameter

It is possible that an artefact repository has a different name from the artefact itself due to artefact naming conventions. For example, scala-js web jar artefacts naming convention is to add a suffix of '_sjs' with the version of the scala-js plugin, i.e. '_sjs0.6'.

### CODE REVIEW

**Comment by SteveSmithTech** [comment]
Thanks @duncancrawford someone will look at this soon

**Comment by duncancrawford** [comment]
Lovely! @agilestevesmith

File: src/main/scala/uk/gov/hmrc/releaser/Releaser.scala [hunk 1]
```diff
@@ -208,10 +209,12 @@ class Releaser(stageDir:Path,
                connectorBuilder:(RepoFlavour) => RepoConnector,
                coordinator:Coordinator){
 
-  def start(artefactName: String, gitRepo:Repo, rcVersion: ReleaseCandidateVersion, targetVersionString: ReleaseVersion): Try[Unit] = {
+  def start(artefactName: String, artefactSuffix : String, gitRepo:Repo, rcVersion: ReleaseCandidateVersion, targetVersionString: ReleaseVersion): Try[Unit] = {
 
-    repositoryFinder(artefactName) flatMap { repo =>
-      val ver = VersionMapping(repo, artefactName, gitRepo, rcVersion, targetVersionString)
+    val artName = s"$artefactName$artefactSuffix"
```
Comment (beyond-code-github) [inline src/main/scala/uk/gov/hmrc/releaser/Releaser.scala:214]: @duncancrawford Silly question... can you not just pass in the artefact name including the suffix to achieve the same result?

**Comment by SteveSmithTech** [comment]
Hi @duncancrawford 

@beyond-code-github and I have looked at your PR. We don't believe it will solve your problem of releasing multiple artifacts associated with the same release candidate version. 

The Releaser is capable of reversioning JARs and POMs associated with a release candidate with a specific version and a name that matches the Bintray and the HMRC repository. If you wish to use JARs, we suggest you publish multiple artifacts at the same version using different classifiers i.e. NAME-VERSION_CLASSIFIER.jar. The Releaser is theoretically capable of supporting that but it would need to be trialled.

**Comment by duncancrawford** [comment]
@agilestevesmith adding a classifier is not the appropriate approach for these types of artefacts. I understand this might be an anomaly but I do need to be able to publish my artefacts. This PR is not about deploying multiple artefacts but about following external & internal artefact naming conventions. I think we need to discuss

**Pull request closed without merging**